Sources/SecureXPC/XPCDecoder.swift

```diff
@@ -71,7 +71,7 @@ enum XPCDecoder {
     }
 }
 
-fileprivate class XPCDecoderImpl: Decoder {
+internal class XPCDecoderImpl: Decoder {
```

Comment:
These types would no longer be able to be `fileprivate`.

I find `XPCEncoder.swift` and `XPCDecoder.swift` pretty hard to navigate, because their contents are so self similar. IDK what your personal taste is like @jakaplan, but I think they would would benefit from refactoring. Perhaps something like:

```
Sources/SecureXPC
├── Decoder
│   ├── XPCDecoder.swift // Now slimmed down
│   ├── XPCDecoderImpl.swift
│   ├── XPCKeyedDecodingContainer.swift
│   ├── XPCSingleValueDecodingContainer.swift
│   └── XPCUnkeyedDecodingContainer.swift
├── Encoder
│   ├── XPCEncoder.swift // Now slimmed down
│   ├── XPCEncoderImpl.swift
│   ├── XPCKeyedEncodingContainer.swift
│   ├── XPCSingleValueEncodingContainer.swift
│   └── XPCUnkeyedEncodingContainer.swift
└── ...
```

WDYT?

Reply:
Opened https://github.com/trilemma-dev/SecureXPC/pull/16/files